File: HugoServer/src/main/java/com/hugo/chat/domain/message/MessageService.java
package com.hugo.chat.domain.message;


import com.hugo.chat.model.message.dto.MessageDTO;

import java.util.Collection;

public interface MessageService {
    MessageDTO createMessage(MessageDTO message, String roomId);

    Collection<MessageDTO> getLatestMessages(String messageId, int amount, String roomId);

    Collection<MessageDTO> getLatestMessages(int amount, String roomId);
}

File: hugo-client/src/services/message/MessageService.ts
import MessageDTO from "./model/MessageDTO";
import {MessageProps} from "../../pages/chat/Messages";

export default interface MessageService {
    getLatestMessages(roomId: string, amount: number): Promise<MessageDTO[]>;

    getMessagesBefore(roomId: string, beforeMessage: string, amount: number): Promise<MessageDTO[]>;

    getMessagesAfter(roomId: string, afterMessage: string): Promise<MessageDTO[]>;

    createMessage(msg: MessageDTO): Promise<MessageDTO>;

    dtoToProps(dtos: MessageDTO[], ownId: string): MessageProps[];
}